metricshub-hardware/src/main/java/org/metricshub/hardware/util/MonitorNameBuilder.java

```diff
@@ -0,0 +1,1020 @@
+
+	private MonitorNameBuilder() {}
+
+	// Enclosure details
+	public static final String HP_OPEN_VMS_COMPUTER = "HP Open-VMS Computer";
+	public static final String HP_TRU64_UNIX_COMPUTER = "HP Tru64 Computer";
+	public static final String HP_UX_COMPUTER = "HP-UX Computer";
+	public static final String IBM_AIX_COMPUTER = "IBM AIX Computer";
+	public static final String LINUX_COMPUTER = "Linux Computer";
+	public static final String MGMT_CARD_ENCLOSURE = "Management Card";
+	public static final String WINDOWS_COMPUTER = "Windows Computer";
+	public static final String NETWORK_SWITCH_ENCLOSURE = "Network Switch";
+	public static final String STORAGE_ENCLOSURE = "Storage System";
+	public static final String SUN_SOLARIS_COMPUTER = "Oracle Solaris Computer";
+	public static final String LOCALHOST_ENCLOSURE = System.getProperty("os.name") + " " + System.getProperty("os.arch");
+
+	// Error messages
+	private static final String HOST_MONITOR_CANNOT_BE_NULL = "hostMonitor cannot be null.";
+	private static final String HOST_TYPE_CANNOT_BE_NULL = "DeviceKind cannot be null.";
+	private static final String ATTRIBUTES_CANNOT_BE_NULL = "Attributes cannot be null.";
+
+	private static final Map<DeviceKind, String> COMPUTE_DISPLAY_NAMES;
+
+	static {
+		final Map<DeviceKind, String> map = new EnumMap<>(DeviceKind.class);
+		for (DeviceKind deviceKind : DeviceKind.values()) {
+			// @formatter:off
+			// @CHECKSTYLE:OFF
+			final String value =
+				switch (deviceKind) {
+					case VMS -> HP_OPEN_VMS_COMPUTER;
+					case TRU64 -> HP_TRU64_UNIX_COMPUTER;
+					case HPUX -> HP_UX_COMPUTER;
+					case AIX -> IBM_AIX_COMPUTER;
+					case LINUX -> LINUX_COMPUTER;
+					case OOB -> MGMT_CARD_ENCLOSURE;
+					case WINDOWS -> WINDOWS_COMPUTER;
+					case NETWORK -> NETWORK_SWITCH_ENCLOSURE;
+					case STORAGE -> STORAGE_ENCLOSURE;
+					case SOLARIS -> SUN_SOLARIS_COMPUTER;
+					default -> null;
+				};
+			// @CHECKSTYLE:ON
+			// @formatter:on
+			map.put(deviceKind, value);
+		}
+
+		COMPUTE_DISPLAY_NAMES = Collections.unmodifiableMap(map);
+	}
+
+	/**
+	 * Checks whether the given string has meaningful content – that is, it is not null,
+	 * empty, or composed solely of whitespace characters.
+	 *
+	 * @param data the {@link String} to be checked
+	 * @return {@code true} if the string has meaningful content; {@code false} otherwise
+	 */
+	protected static boolean hasMeaningfulContent(final String data) {
+		return data != null && !data.trim().isEmpty();
+	}
+
+	/**
+	 * Trims unwanted characters from the specified name. This method removes commas,
+	 * empty parentheses, and replaces multiple whitespace characters with a single space,
+	 * then trims any leading or trailing whitespace.
+	 *
+	 * @param name the {@link String} name to be trimmed
+	 * @return the trimmed name
+	 */
+	static String trimUnwantedCharacters(final String name) {
+		return name.replace(",", "").replace("()", "").replaceAll(WHITE_SPACE_REPEAT_REGEX, WHITE_SPACE).trim();

```

Comment:
`WHITE_SPACE_REPEAT_REGEX` should be compiled as a `Pattern` rather than used as a `String`.